models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -415,4 +415,71 @@ public function loadLazyBrickField(DataObject\Objectbrick\Data\AbstractData $ite
         }
     }
+
+    /**
+     * @internal trigger deprecation error when a relation is passed multiple times, remove in Pimcore 7
+     *
+     * @param array|null $data
+     * @param DataObject\Concrete|DataObject\Localizedfield|DataObject\Objectbrick\Data\AbstractData|\Pimcore\Model\DataObject\Fieldcollection\Data\AbstractData $object
+     * @param array $params
+     *
+     * @return array
+     *
+     * @throws \Exception
+     */
+    public function filterMultipleAssignments($data, $object, $params)
+    {
+        if(!is_array($data) || count($data) < 2) {
+            return $data;
+        }
+
+        if (!method_exists($this, 'getAllowMultipleAssignments') || !$this->getAllowMultipleAssignments()) {
+                $relationItems = [];
+                $objectId = null;
+                $fieldName = $this->getName();
+
+                if ($object instanceof DataObject\Concrete) {
+                    $objectId = $object->getId();
+                } elseif (
+                    $object instanceof DataObject\Fieldcollection\Data\AbstractData ||
+                    $object instanceof DataObject\Localizedfield ||
+                    $object instanceof DataObject\Objectbrick\Data\AbstractData
+                ) {
+                    $objectFromContainer = $object->getObject();
+                    if ($objectFromContainer) {
+                        $objectId = $objectFromContainer->getId();
+                    }
+                }
+
+                foreach ($data as $item) {
+                    $elementHash = null;
+                    if ($item instanceof DataObject\Data\ObjectMetadata || $item instanceof DataObject\Data\ElementMetadata) {
+                        if ($item->getElement() instanceof Element\ElementInterface) {
+                            $elementHash = Element\Service::getElementHash($item->getElement());
+                        }
+                    } elseif ($item instanceof Element\ElementInterface) {
+                        $elementHash = Element\Service::getElementHash($item);
+                    }
+
+                    if($elementHash === null) {
+                        $relationItems[] = $item; //do not filter if element hash fails
+                    } elseif (!isset($relationItems[$elementHash])) {
+                        $relationItems[$elementHash] = $item;
+                    } elseif (isset($relationItems[$elementHash])) {
+                        @trigger_error(
+                            'Passing relations multiple times is deprecated since version 6.5.2 and will throw exception in 7.0.0, tried to assign ' . $elementHash
+                            . ' multiple times in field' . $fieldName . ' of object id: ' . $objectId,
+                            E_USER_DEPRECATED
+                        );
+                    }
+                }
+
+                if(count($relationItems) !== count($data)) {
+                    $this->setDataToObject($relationItems, $object, $params);
```

Comment:
maybe change to setDataToObject(array_values($relationItems), ...)
but ... I am not sure if we should change the array keys anyway.
calling the setter, save it and then calling the getter will probably not return the same thing

Reply:
changed it setDataToObject(array_values($relationItems). Anyway passing associative array for relation types is not a valid structure.